### PR TITLE
KZL 174 - Update README for SDK V6

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do?

This PR update the README to briefly explain how to use the SDK V6 until the documentation is available.

### How should this be manually tested?

  - Step 1 : Read the readme, try the code example

### Other changes
  
 - update deploy script (the one who build the `dist/`) to support the `6-beta` branch

### Boyscout

 - cosmetics on `.travis.yml`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
...
 // with the classic require...
-let Kuzzle = require('kuzzle-sdk/dist/kuzzle.js')
+const { Kuzzle } = require('kuzzle-sdk/dist/kuzzle.js')
 // ... or with the new import directive.
-import Kuzzle from 'kuzzle-sdk/dist/kuzzle.js'
+import { Kuzzle } from 'kuzzle-sdk/dist/kuzzle.js'
+```
+
+## Protocols used
+Actuellement, le SDK supporte nativement 3 protocols: `http`, `websocket` et `socketio`.
```
Comment (ballinette) [inline README.md]: to be translated into English... ;)

File: README.md [hunk 2]
```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
...
+const { Kuzzle } = require('kuzzle-sdk/dist/kuzzle.js')
 // ... or with the new import directive.
-import Kuzzle from 'kuzzle-sdk/dist/kuzzle.js'
+import { Kuzzle } from 'kuzzle-sdk/dist/kuzzle.js'
+```
+
+## Protocols used
+Actuellement, le SDK supporte nativement 3 protocols: `http`, `websocket` et `socketio`.  
+
+The main reason behind this is that while Socket.IO offers better compatibility with older web browsers, our raw WebSocket implementation is about 20% faster
```
Comment (ballinette) [inline README.md]: nitpicking: remove `The main reasion behind this is that`

just be decalarative, something like that:
```
Websocket and Socket.IO protocols implement the whole Kuzzle API, while HTTP protocol does not implement realtime features (rooms and subscriptions)
While Socket.IO offers better compatibility with older web browsers, our raw WebSocket implementation is about 20% faster
```

File: README.md [hunk 3]
```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
...
+
+Which protocol is used when you connect to Kuzzle depends on multiple factors:
+
+#### NodeJS
+
+The protocol used is always raw WebSocket for `websocket`.
```
Comment (ballinette) [inline README.md]: not true anymore. We can now choose btw http and websocket (and even socket.io if we want, whereas it is not a pertinent choice), or implement our custom protocol if needed.

File: README.md [hunk 4]
```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
...
+
+#### Web Browsers
+
+If you choose `websocket`, the SDK will first try to use raw WebSocket to connect to Kuzzle. If the web browser does not support this protocol, then the SDK falls back to Socket.IO.
```
Comment (ballinette) [inline README.md]: not true anymore: the application must now implement the fallback if we need it.
(maybe give a short example for that.)

File: README.md [hunk 5]
```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
...
+  await kuzzle.connect();
+  const serverTime = await kuzzle.server.now();
+
+  console.log(serverTime);
+} catch (error) {
+  console.error(error);
+}
+```
+
+### ISO with the Kuzzle API
```
Comment (ballinette) [inline README.md]: open suggestion: we could also add an sample code with `kuzzle.query` method.

File: README.md [hunk 6]
```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
...
+
+### Promise based
+
+Each SDK method returns a Promise resolving on the result of the API return (the `result` property described in the API documentation).  
+
+For example, for the action `create` of the controller `collection` ([collection#create](https://docs.kuzzle.io/api-documentation/controller-collection/create)), the property `result` contains `{ "acknowledged": true} `. This is therefore what will be returned by the SDK method if successful.
+
+Any errors must be caught either at the end of the Promise chain, or by using `async/await` and a `try...catch`.
+
+Translated with www.DeepL.com/Translator
```
Comment (ballinette) [inline README.md]: ???

File: README.md [hunk 7]
```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
...
+const documentBody = { hello: 'world' };
+kuzzle.document.create('my-index', 'my-collection', 'my-uniq-id', documentBody, options)
+```
+
+The parameters of each method differ according to the parameters expected in the API.  
+To get the details of the parameters of each method, it is necessary for the moment to see the code of each controller on [Github](https://github.com/kuzzleio/sdk-javascript/tree/6-beta/src/controllers).
+
+### Promise based
+
+Each SDK method returns a Promise resolving on the result of the API return (the `result` property described in the API documentation).
```
Comment (scottinet) [inline README.md]: All SDK methods return a promise resolving the `result` part of Kuzzle API responses. If an error occurs, the promise is rejected with an `Error` object embedding the `error` part of the API response.

File: README.md [hunk 8]
```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
...
+The parameters of each method differ according to the parameters expected in the API.  
+To get the details of the parameters of each method, it is necessary for the moment to see the code of each controller on [Github](https://github.com/kuzzleio/sdk-javascript/tree/6-beta/src/controllers).
+
+### Promise based
+
+Each SDK method returns a Promise resolving on the result of the API return (the `result` property described in the API documentation).  
+
+For example, for the action `create` of the controller `collection` ([collection#create](https://docs.kuzzle.io/api-documentation/controller-collection/create)), the property `result` contains `{ "acknowledged": true} `. This is therefore what will be returned by the SDK method if successful.
+
+Any errors must be caught either at the end of the Promise chain, or by using `async/await` and a `try...catch`.
```
Comment (scottinet) [inline README.md]: Any error

It doesn't seem useful to me to explain how a promise should be handled, I think that my proposal above is enough to document how errors are returned by SDK methods

File: README.md [hunk 9]
```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
...
+// Without async/await
+kuzzle.connect()
+  .then(() => {
+    // You are now connected to your Kuzzle instance.
+    return kuzzle.server.now();
+  })
+  .then(serverTime => console.log(serverTime))
+  .catch(error => console.error(error));
+
+// With async/await
```
Comment (scottinet) [inline README.md]: What's the purpose of proving an async/await example?

File: README.md [hunk 10]
```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
...
+
+Translated with www.DeepL.com/Translator
+
+```js
+const mapping = {
+  properties: {
+    hello: { type: 'text' }
+  }
+};
+// Without async/await
```
Comment (scottinet) [inline README.md]: ditto: I don't think that this async/await example adds anything interesting to our documentation, we should stick to raw promises examples.

File: README.md [hunk 11]
```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
...
+kuzzle.collection.create('my-index', 'my-collection', mapping)
+  .then(result => console.log('Success'))
+  .catch(error => console.error(`Hu oh, we've got some error: ${error.message}`));
+
+// With async/await
+try {
+  const result = await kuzzle.collection.create('my-index', 'my-collection', mapping);
+  // result contain { "acknowledged": true }
+  console.log('Success');
```
Comment (scottinet) [inline README.md]: `console.log(result);` seems more appropriate to me

File: README.md (hunk 6 above)
Comment (Aschen) [reply README.md]: :sweat_smile:

File: README.md [hunk 12]
```diff
@@ -84,9 +82,100 @@ But you'll still need to pick the built version (which ships with the package).
...
-import Kuzzle from 'kuzzle-sdk/dist/kuzzle.js'
+import { Kuzzle } from 'kuzzle-sdk/dist/kuzzle.js'
+```
+
+## Protocols used
+Actuellement, le SDK supporte nativement 3 protocols: `http`, `websocket` et `socketio`.  
+
+The main reason behind this is that while Socket.IO offers better compatibility with older web browsers, our raw WebSocket implementation is about 20% faster
+
+Which protocol is used when you connect to Kuzzle depends on multiple factors:
```
[the remark this answers is not in the document]
Comment (Aschen) [reply README.md]: I removed this sentence

File: README.md (hunk 8 above)
Comment (Aschen) [reply README.md]: I feel more confident when I see example along the documentation but if you think it's too much I can remove that part.

File: README.md (hunk 9 above)
Comment (Aschen) [reply README.md]: Same reason as above

File: README.md (hunk 10 above)
Comment (Aschen) [reply README.md]: async/await feel more modern than then/catch. With the rise of async/await over raw promises, I think more and more people will be more comfortable with async/await syntax.

File: README.md [hunk 13]
```diff
@@ -10,37 +10,37 @@ A backend software, self-hostable and ready to use to power modern apps.
 
 You can access the Kuzzle repository on [Github](https://github.com/kuzzleio/kuzzle)
 
+This is the documentation for the beta-6 SDK. We are currently refactoring in deep our documentation to support many versions of each SDK. You can see it [here](https://docs-v2.kuzzle.io/sdk-reference/kuzzle/constructor/) but notice that this is an beta version of the documentation.
```
Comment (ballinette) [inline README.md]: `s/this is an beta version/this is a beta version/`

**Pull request merged**